Symfony/CS/Fixer/Contrib/EregToPregFixer.php

```diff
@@ -0,0 +1,173 @@
+                // 0 => function name,
+                // 1 => bracket "("
+                // 2 => quoted string passed as 1st parameter
+                $match = array_keys($match);
+
+                // advance tokenizer cursor
+                $currIndex = $match[2];
+
+                // ensure it's a function call (not a method / static call)
+                $prev = $tokens->getPrevMeaningfulToken($match[0]);
+                if (null === $prev || $tokens[$prev]->isGivenKind(array(T_OBJECT_OPERATOR, T_DOUBLE_COLON))) {
+                    continue;
+                }
+
+                // ensure the first parameter is just a string (e.g. has nothing appended)
+                $next = $tokens->getNextMeaningfulToken($match[2]);
+                if (null === $next || !$tokens[$next]->equalsAny(array(',', ')'))) {
+                    continue;
+                }
+
+                // convert to PCRE
+                $string = substr($tokens[$match[2]]->getContent(), 1, -1);
+                $quote = substr($tokens[$match[2]]->getContent(), 0, 1);
+                $delim = $this->getBestDelimiter($string);
+                $preg = $delim.addcslashes($string, $delim).$delim.'D'.$map[2];
+
+                // check if the preg is valid
+                if (!$this->checkPreg($preg)) {
+                    continue;
+                }
+
+                // modify function and argument
+                $tokens[$match[2]]->setContent($quote.$preg.$quote);
+                $tokens[$match[0]]->setContent($map[1]);
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Replace deprecated ereg regular expression functions with preg. Warning! This could change code behavior.';
+    }
+
+    /**
+     * Check the validity of a PCRE
+     *
+     * @param string $pattern the regular expression
+     *
+     * @return bool
+     */
+    private function checkPreg($pattern)
+    {
+        return false !== @preg_match($pattern, '');
+    }
+
+    /**
+     * Get the delimiter that would require the least escaping in a regular expression.
+     *
+     * @param string $pattern the regular expression
+     *
+     * @return string the preg delimiter
+     */
+    private function getBestDelimiter($pattern)
+    {
+        // try do find something that's not used
+        $delimiters = array();
+        foreach (self::$delimiters as $k => $d) {
+            if (false === strpos($pattern, $d)) {
+                return $d;
+            }
+
+            $delimiters[$d] = array(substr_count($pattern, $d), $k);
+        }
+
+        // return the least used delimiter, using the position in the list as a tie breaker
+        uasort($delimiters, function ($a, $b) {
+            if ($a[0] === $b[0]) {
+                if ($a[1] === $b[1]) {
+                    return 0;
```

Comment:
BTW, when there is only one author in PR you don't need to squash. We can do this automatically during merging